dns-global-exclusive-tls.ks.in

```diff
@@ -0,0 +1,81 @@
+# Test the e2e functionality of global exclusive encrypted DNS functionality
+# Partial tests of reqiured features are in 'certificate' and 'dns-global-bootopts'
+# tests.
+
+# Use defaults, but no network
+%ksappend common/common.ks
+%ksappend repos/default.ks
+%ksappend l10n/default.ks
+%ksappend users/default.ks
+%ksappend storage/default.ks
+%ksappend payload/default_packages.ks
+
+%certificate --filename tls-ca-bundle.pem --dir /etc/pki/dns/extracted/pem
+%ksappend data/edns-idmops.psi.redhat.com.crt
+%end
+
+# Reboot the installed system.
+reboot
+
+# Validate on the first boot.
+%ksappend validation/success_on_first_boot.ks
+
+
+%post --nochroot
+
+# Check the presence on installed system
+if [ ! -f /etc/pki/dns/extracted/pem/tls-ca-bundle.pem ]; then
+    echo "*** tls-ca-bundle.pem not found in /etc/pki/dns/extracted/pem in installer environment" >> /root/RESULT
+fi
+
+HOSTNAME=fedoraproject.org
+getent hosts ${HOSTNAME}
+if [[ $? -ne 0 ]]; then
+    echo "*** Failed check: name resolution on ${HOSTNAME} failed" >> /root/RESULT
+fi
+
+%end
+
+%post
+
+# Check the presence on installed system
+if [ ! -f /etc/pki/dns/extracted/pem/tls-ca-bundle.pem ]; then
+    echo "*** tls-ca-bundle.pem not found in /etc/pki/dns/extracted/pem on installed system" >> /root/RESULT
+fi
+
+%ksappend validation/success_if_result_empty.ks
+
+HOSTNAME=fedoraproject.org
+getent hosts ${HOSTNAME}
+if [[ $? -ne 0 ]]; then
+    echo "*** Failed check: name resolution on ${HOSTNAME} in %post script failed" >> /root/RESULT
+fi
+
+%end
+
+# Set up the actual test.
+%post
+
+# Write the code to run after reboot. It lists files with wrong SELinux contexts.
+cat > /usr/libexec/kickstart-test.sh << 'EOF'
+
+HOSTNAME=fedoraproject.org
+succeeded="no"
+i=0
+while [ $i -le 15 ] && [ ${succeeded} != "yes" ];
```

Comment:
Shouldn't this work immediately, without trying several attempts? Is this test started before network is up and running?

Reply:
The test ([our test service](https://github.com/rhinstaller/kickstart-tests/blob/main/fragments/shared/validation/success_on_first_boot.ks)) does not wait for network being online.
Maybe we could wait for the network-online target here in the script but this timeout solution seems more robust in a sense to me.

Reply:
Added